lib/scrabble.rb

```diff
@@ -1,4 +1,104 @@
 module Scrabble
-  # require your gems and classes here
-  # require_relative 'scrabble/whatevs'
+require "awesome_print"
+
+  class Scrabble
+    #___________________________________________________________________________
+    # Constants
+
+    SCORE_1 =  ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"]
+    SCORE_2 =  ["D", "G"                                        ]
+    SCORE_3 =  ["B", "C", "M", "P"                              ]
+    SCORE_4 =  ["F", "H", "V", "W", "Y"                         ]
+    SCORE_5 =  ["K"                                             ]
+    SCORE_8 =  ["J", "X"                                        ]
+    SCORE_10 = ["Q", "Z"                                        ]
+    #___________________________________________________________________________
+    # Class method score
+
+    def self.score(word)
+
+      # the complex if statement verifies that argument in method:
+      # it's not a number and is not string "0"
+      # length isn't bigger than 7 letters
+      # there is no spaces
+
+      if (word.to_i.class == Fixnum && (word.to_i != 0 || word == "0" || word == 0)) || (word.length > 7 || word.split() != [word])
+          return "Error!"
+        else
+          word = word.upcase
+       end
+
+      # turnig word into array of letters
+      # matching every letter of the word with the letters in Constants arrays, to get the score
+
+      word = word.split("")
```

Comment:
I love the clean structured approach of associating characters to points.  I will challenge you to think of a data structure or a combination of data structures that could help change this combination of if else statements into a more standalone query.  Hint: Think of scores as a mapping of a words letters to a numeric value.